src/material/badge/badge.ts

```diff
@@ -238,15 +254,37 @@ export class MatBadge extends _MatBadgeBase implements OnInit, OnDestroy, CanDis
       this._badgeElement.textContent = newContentNormalized;
     }
 
+    if (!this._isHostInteractive()) {
+      if (!this._inlineBadgeDescription) {
+        this._inlineBadgeDescription = this._renderer.createElement('span') as HTMLSpanElement;
```

Comment:
We don't need to go through the renderer for this. The badge uses the renderer so view encapsulation works correctly, but this element doesn't need it.